### ISSUE TITLE
Can't connect using server name

### ISSUE THREAD

**Issue by hartmms** [owntracks/android#418]
I run a private MQTT broker (mosquitto) behind a firewall on a home server.  My network router/firewall has the 8883 port forwarded to the server IP (TCP forwarding). I also use TLS.  I can't get owntracks to connect when using my dynamic DNS name of my server. The status is always "Error: Connection refused".   If I specify my server IP address instead of the name, it will connect perfectly. When I push the "publish/up" icon on the main screen, I can see my publish happen on my mosquitto logs.   This tells me I have my TLS and authentication settings correct. 

Something I noticed was when I click the "checkmark" symbol on the settings activity to accept my hostname, the "x" next to it stays gray. When I specify the IP address and click "checkmark" (and it connects successfully),  the "x" changes from gray to white. 

Perhaps I'll try to compile my own version of owntracks so I can debug it with Android Studio.  I see some useful LOG debug entries in the source code that might help me track it down. 

More data about my setup:

- the certificate used for mosquitto is bound to my domain name, not my IP address
- I have used owntracks with this mosquitto broker in the past with no issues (TLS and all)
- configuration:
```
  Host:
    host:  <name>.duckdns.org
    port: 8883
    websockets: off
  Identification:
    authentication: on
    username: <user>
    password: <pass>
    device ID: 6p
    Tracker ID: 6p
   Security:   
     TLS: on
     CA cert:  ca.pem
     client cert: none
     private key: none
   parameters:
     clean session: on
     keepalive: 600
```

**Comment by binarybucks** [comment]
Connection refused means that no broker is listening at at the address
you specified. Your dynamic DNS entry most probably points to the
wrong IP.

> Am 17.10.2016 um 04:19 schrieb Michael <notifications@github.com>:
>
> Connection refused

**Comment by hartmms** [comment]
I had thought about that before I posted, and I don't think that is the issue.  

1) I can connect to other services I have running on my server when I use the name  (a FQDM).
2) a pen test website can see the <name>.duckdns.org:8883 port as being open. 

Believe me, this issue definitely screams configuration or IP address resolution issue. But I think I have tested all possibilities that eliminate these as possibilities. 

I'll see if my local compile with debug can shed any light to the issue.
